src/ui/public/vis/agg_configs.js

```diff
@@ -47,6 +47,23 @@ export default function AggConfigsFactory(Private) {
     }
   }
 
+  /**
+   * Data-by-data comparison of this Aggregation
+   * Ignores the non-array indexes
+   * @param aggConfigs an AggConfigs instance
+   */
+  AggConfigs.prototype.jsonDataEquals = function (aggConfigs) {
+    if (aggConfigs.length !== this.length) {
+      return false;
+    }
+    for (let i = 0; i < this.length; i += 1) {
+      if (!_.isEqual(aggConfigs[i].toJSON(),this[i].toJSON())) {
```

Comment:
There should be a space here between the two arguments to isEqual.  I'm actually surprised that the linter allowed this...

Also, is property order guaranteed by the toJSON function?  If not, then this isn't really a safe way to verify equality.  To be truly guarantee-able, we need to have unit tests on toJSON to verify that behavior.